Jenkins_jobs/CreateNewNode.groovy

```diff
@@ -16,20 +16,88 @@ node {
         String[] machines = params.machineNames.split(",")
         String[] machineIPs = params.machineIPs.split(",")
         String[] labels = params.labelStrings.split(",")
+        def FILES = []
+        git url: "git@github.ibm.com:Xiaoxiao-Wang/infra-can.git", credentialsId: Constants.SSH_CREDENTIAL_ID, branch:"INI"
+        FILES = sh (
+                    script: "cat ansible/playbooks/Infrastructure/dynamic_inventory.ini",
+                    returnStdout: true
+                ).trim()
+        def FILES_LIST = FILES.split("\\n")
+        echo "${FILES_LIST}"
+        for (machine in FILES_LIST){ 
+            def EXISTED = false;
+            def emptyMap = [:];
+            def halfmachine = "";
+            if (machine =~ /[^\s]+ \s*(\S+)\s*=\s*(.*)?(?:#|$)\S*/) {
+            if (machine.contains("#")) {
+                def location = machine.indexOf("#")
+                halfmachine = machine.substring(location + 1,machine.length())
+            } else {
+                halfmachine = ""
+            }
+            def splitLines = halfmachine.split(',')
+            splitLines.each {
+                if (it != "" && it != null){
+                    String[] parts = it.split("=",2);
+                    emptyMap.put(parts[0],parts[1])
+                }
+            }
+                for (aSlave in hudson.model.Hudson.instance.slaves) {
+                    if (machine.contains(aSlave.name)){
+                    EXISTED = true;
+
+                    if (emptyMap.get("Labels") != null && emptyMap.get("Labels") != aSlave.getLabelString()) {
+                        aSlave.setLabelString(emptyMap.get("Labels"))
+                    }
+
+                    if (emptyMap.get("Executor") != null && emptyMap.get("Executor") != aSlave.getNumExecutors()) {
+                        aSlave.setNumExecutors(emptyMap.get("Executor").toInteger())
+                    }
+
+                    if (emptyMap.get("Description")!= null && emptyMap.get("Description") != aSlave.getDescriptor()){
+                        aSlave.setNodeDescription(emptyMap.get("Description"))
+                    }
+                    break;
+                }
+                }
+                def name = "";
+                if (!EXISTED){
+                    machine.find (/\S+.ibm.com/){
+                        def location = it.indexOf(".")
+                        it = it.substring(0,location)
+                        name = it
+                    }
+                    machines = machines + "${name}"
```

Comment:
Can you please add comments here with an sample example and maybe a one line explanation

Reply:
commented